src/lib/storage/lz4_segment.cpp

```diff
@@ -454,6 +455,16 @@ std::optional<CompressedVectorType> LZ4Segment<T>::compressed_vector_type() cons
   return std::nullopt;
 }
 
+template <>
+std::optional<CompressedVectorType> LZ4Segment<pmr_string>::compressed_vector_type() const {
```

Comment:
Isn't this always `CompressedVectorType::SimdBp128` with your change below?

Reply:
It is, but whenever somebody feels like changing to SimdBp128 or making the decision configurable, the method should still be good.

Reply:
add a comment

Reply:
Added.